javalib/src/main/scala/java/io/File.scala

```diff
@@ -494,9 +494,9 @@ object File {
 
     // Find the next separator
     var i = start
-    while (i < strlen(path) && path(i) != separatorChar) i += 1
```

Comment:
This is scaring me a bit, there can be some pretty wrong behavior if users forget to convert a `Word` to an `Int` before doing comparisons. Is there some way that we could make equality work here (and stop scalac from complaining that the comparison won't work)?

Reply:
So Scala has "cooperative" equality where types which store the same numeric value will compare equals (i.e., `1.toByte == 1.toLong`). We implement it using special `scala_==` methods. For example here is implementation for [`java.lang.Integer`](https://github.com/scala-native/scala-native/blob/master/javalib/src/main/scala/java/lang/Integer.scala#L47).

The question still stands if we should extend it to unsigned and/or word types.

Reply:
I think it's best to move this to dedicated issue and leave this PR as-is without cooperative equality between words and normal numbers.